node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java

```diff
@@ -70,14 +66,14 @@ public Autoscaler(MetricsDb metricsDb, NodeRepository nodeRepository) {
         Optional<Double> cpuLoad    = clusterTimeseries.averageLoad(Resource.cpu);
         Optional<Double> memoryLoad = clusterTimeseries.averageLoad(Resource.memory);
         Optional<Double> diskLoad   = clusterTimeseries.averageLoad(Resource.disk);
-        if (cpuLoad.isEmpty() || memoryLoad.isEmpty() || diskLoad.isEmpty()) return Optional.empty();
+        if (cpuLoad.isEmpty() || memoryLoad.isEmpty() || diskLoad.isEmpty()) return Advice.none();
         var target = ResourceTarget.idealLoad(cpuLoad.get(), memoryLoad.get(), diskLoad.get(), currentAllocation);
 
         Optional<AllocatableClusterResources> bestAllocation =
                 allocationOptimizer.findBestAllocation(target, currentAllocation, limits, exclusive);
-        if (bestAllocation.isEmpty()) return Optional.empty();
-        if (similar(bestAllocation.get(), currentAllocation)) return Optional.empty();
-        return bestAllocation;
+        if (bestAllocation.isEmpty()) return Advice.dontScale();
+        if (similar(bestAllocation.get(), currentAllocation)) return Advice.dontScale();
```

Comment:
Shouldn't this return `currentAllocation`?

In the case this is meant to fix:
1. cfg1 calculates a new target, `t1`, and deploys with it
2. cfg2 decides that the new target, `t2`, which is not sufficiently different from the current, so it returns empty, which means that target `t1` is cleared
3. On the next deploy, on any cfg: There is no target and the nodes provisioned with `t1` are not yet active, so it will revert back to the previous target (per `NodeRepositoryProvsioner::currentResources`)

Reply:
Current isn't the current target but the current allocation (active nodes).